frontend/src/components/InputForm/InputForm.tsx

```diff
@@ -42,36 +42,31 @@ const InputForm: React.FC<InputFormProps> = ({
   const [mode, setMode] = useState("Linear Orders");
 
   const textareaOnBlur = () => {
-    if (mode !== "Linear Orders") return;
-
-    const upsilon = textareaValue
-      .split("\n")
-      .map((line) => line.trim())
-      .filter((line) => line !== "");
-    if (upsilon.length > 0) {
-      setSize(upsilon[0].length);
+    if (mode == "Linear Orders") {
+      const upsilon = textareaValue
+        .split("\n")
+        .map((line) => line.trim())
+        .filter((line) => line !== "");
+      if (upsilon.length > 0) {
+        setSize(upsilon[0].length);
+      }
+    } else if (mode == "Poset") {
+      const numbers = textareaValue
+        .split("\n")
+        .map((line) => line.trim())
+        .filter((line) => line !== "")
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        .map(([num1, _, num2]) => [parseInt(num1), parseInt(num2)])
+        .flat();
+      if (numbers.length > 0) {
```

Comment:
what if 
```suggestion
      if (numbers.length > 0 && size < Math.max(...numbers)) {
```
so that it will not interfere the user if the user wants a larger size than Math.max(...numbers)

Reply:
This would undo the pending change done in InputForm.test.tsx line 94.

3 -> 4

Reply:
Ahh oo nga. You're right.